validphys2/src/validphys/n3fit_data.py

```diff
@@ -97,9 +100,17 @@ def tr_masks(data, replica_trvlseed):
         # We do this so that a given dataset will always have the same number of points masked
         trmax = int(ndata * frac)
         if trmax == 0:
-            # If that number is 0, then get 1 point with probability frac
-            trmax = int(rng.random() < frac)
-        mask = np.concatenate([np.ones(trmax, dtype=bool), np.zeros(ndata - trmax, dtype=bool)])
+            if parallel_models and len(replicas) > 1:
```

Comment:
```suggestion
            if parallel_models:
```

I'd do it like this for parallel models regardless of the number of parallel models, otherwise reproducibility becomes (even more) complicated.